Application/EdFi.Ods.WebApi/StartupFactory.cs

```diff
@@ -0,0 +1,42 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Linq;
+using EdFi.Ods.Api.Startup;
+using Microsoft.AspNetCore.Hosting;
+using System;
+
+namespace EdFi.Ods.WebApi
+{
+    public class StartupFactory(string startupClassName)
+    {
+        private readonly string _startupClassName = startupClassName;
+        private const string MissingStartupClassMessage =
+            "Could not find a startup class named '{0}' that derives from OdsStartupBase.";
+
+        public OdsStartupBase Create(WebHostBuilderContext webHostBuilderContext)
+        {
+            var startupType = FindStartupClassType(_startupClassName);
+            var startup = (OdsStartupBase)Activator.CreateInstance(startupType, webHostBuilderContext.Configuration);
+            return startup;
+        }
+
+        private Type FindStartupClassType(string startupClassName)
+        {
+            var startupType = AppDomain.CurrentDomain.GetAssemblies()
+                .Where(assembly => assembly.GetName().Name?.StartsWith("EdFi.") ?? false)
+                .SelectMany(assembly => assembly.GetTypes())
+                .Where(t => typeof(OdsStartupBase).IsAssignableFrom(t) && t.Name == startupClassName)
```

Comment:
Without digging, I'm not sure how expensive the `IsAssignableFrom` method is, but I know that string comparison should be very fast. Because of C#'s behavior in short circuiting conditional logic once part of the criteria is false, perform the string check first, and change it to use the `EqualsIgnoreCase` method while you're at it. So this line becomes:
```
.Where(t => t.Name.EqualsIgnoreCase(startupClassName) && typeof(OdsStartupBase).IsAssignableFrom(t))
```